lib/config/folderconfiguration.go

```diff
@@ -181,14 +181,16 @@ func (f *FolderConfiguration) DeviceIDs() []protocol.DeviceID {
 	return deviceIDs
 }
 
-func (f *FolderConfiguration) prepare(myID protocol.DeviceID, existingDevices map[protocol.DeviceID]bool) {
+func (f *FolderConfiguration) prepare(myID protocol.DeviceID, existingDevices map[protocol.DeviceID]*DeviceConfiguration) {
 	// Ensure that
 	// - any loose devices are not present in the wrong places
 	// - there are no duplicate devices
 	// - we are part of the devices
+	// - folder is not shared in trusted mode with an untrusted device
 	f.Devices = ensureExistingDevices(f.Devices, existingDevices)
 	f.Devices = ensureNoDuplicateFolderDevices(f.Devices)
 	f.Devices = ensureDevicePresent(f.Devices, myID)
+	f.Devices = ensureNoUntrustedTrustingSharing(f, f.Devices, existingDevices)
```

Comment:
Alternatively we could return an error to reject the config change as invalid. Then an external config tool using the API would get a notification immediately, without having to check the errors endpoint.

Reply:
Hm, possibly, but this code also runs for loading the config from disk. We don't want to error out on startup, right?

Reply:
We could add an optional param `WithCleanup` to `config.Load`: When used we immediately fix it in place (like now) instead of returning an error.

Reply:
Possibly, but why now and why not do the same for all the other things we fix in various prepare steps?